devicetypes/smartthings/zigbee-window-shade.src/zigbee-window-shade.groovy

```diff
@@ -94,37 +101,60 @@ def parse(String description) {
                     sendEvent(name: "level", value: zigbee.convertHexToInt(liftmap.value))
                 }
             }
+        } else if (isWistarDevice() && descMap?.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER && descMap.value) {
```

Comment:
Rather than checking for a particular model which likely won't scale well, you could add functions like `usesLevelControl` and/or `usesWindowCovering`. Then when more manufacturers are added you just add update that one function rather than having to update multiple parts of the code.

Reply:
it's a good idear

Reply:
@BYDengS I believe what @tpmanley meant was to leave the code in `parse()` as is, but instead of having a function like `isWistar()` to have `usesLevelControl()`.

Reply:
@BYDengS So I guess really in this case, `isWistarDevice()` would become `usesLevelControl()` in the code before your last commit.

Reply:
@dkirker ah,I misunderstood.